iorgen/parser_rust.py

```diff
@@ -17,15 +18,19 @@
     "yield"
 ]
 
+# KEYWORDS extended with the list of identifiers used by the generated code
+EXTENDED_KEYWORDS = KEYWORDS + [
+    "buffer", "read_line", "std", "Box", "String", "Vec"
+]
+
+# Conventional indentation token
 INDENTATION = "    "
```

Comment:
You did not use this in most of your code. It’s not a deal breaker because we will probably always use 4 spaces for rust, but you should at least add in your comment that it will not be used everywhere in the code, and that changing the content of `INDENTATION` will not work as expected.

Reply:
Oh! That's what I forgot to ask about in my last comment!

Using format strings makes it easy to generate pieces of codes with a quite nice indentation, but I chose to not consistently use the `INDENTATION` constant. If you think this is something we should keep, this can probably be dealt with a fair style such as:

```python
__ = INDENTATION

read_line = [
    u"fn read_line(mut buffer: &mut String) -> &str {",
    f"{__}buffer.clear();",
    f"{__}std::io::stdin()",
    f"{__}{__}.read_line(&mut buffer)",
    f'{__}{__}.expect("impossible to read a new line");',
    f"{__}buffer.trim_end()",
    u"}",
]
```

Reply:
That’s a good idea, but I’m not sure `__` is a variable name that should be used. Maybe something like `tab`